modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java

```diff
@@ -233,6 +236,12 @@ public PainlessMethod lookupFunctionalInterfacePainlessMethod(Class<?> targetCla
         return targetPainlessClass.functionalInterfaceMethod;
     }
 
+    public Class<?> typeOfScriptToInjectForMethod(String methodName, int methodArity) {
```

Comment:
So, I think this would fit better if the script was a parameter of the whitelisted method. Then this method wouldn't be need. It seems awkward to me to have what is specialized context information in the lookup that doesn't really know anything about contexts.